### DISCUSSION TITLE
Rules are only applied from the last override?

### DISCUSSION THREAD

**Discussion by AnderssonPeter** [eslint/eslint#15861] [Q&A]
I have multiple `overrides` that target different file subsets, but some overlap.
But eslint only seems to use the last matched override, thereby ignoring some of my rules.

I have the following config:
```json
{
  "root": true,
  "overrides": [
    {
      "files": [
        "**/map/**/*.ts"
      ],
      "rules": {
        ...
      }
    },
    {
      "files": [
        "**/infrastructure/**/*.ts"
      ],
      "rules": {
        ...
      }
    }
  ]
}
```

But the following file `map\infrastructure\map.models.ts` only gets the rules applied from the `"**/infrastructure/**/*.ts"`.
Is there someway to apply both?

**Comment by mdjermanovic** [comment]
All overrides that match the file should apply. Can you provide a repro repo where it doesn't seem to work that way?

**Reply by AnderssonPeter** [reply]
Sorry for the late reply.

I have recreated the issue here
[test.zip](https://github.com/eslint/eslint/files/8670323/test.zip)

If you run `ng lint`, then you switch the the overrides (the one with foo and bar), it will give you different linting in the app.module.ts

When bar is first i get the following:
```bash
C:\projects\recreate-issue\test\src\app\app.module.ts
  1:1  error  '@angular/core' import is restricted from being used by a pattern. Foo              no-restricted-imports
  2:1  error  '@angular/platform-browser' import is restricted from being used by a pattern. Foo  no-restricted-imports
```
When Foo is first i get the following:
```bash
C:\projects\recreate-issue\test\src\app\app.module.ts
  2:1  error  '@angular/platform-browser' import is restricted from being used by a pattern. Bar  no-restricted-imports
```

So the order seems to matter does seem to matter and results in some rules just being outright ignored?

**Reply by AnderssonPeter** [reply]
If you prefer a github repo you can find it here https://github.com/AnderssonPeter/eslint-issue

**Reply by mdjermanovic** [reply]
`no-restricted-imports` is a rule, `["error", { "patterns": [...] }]` is a configuration for that rule.

If there's same rule (the `no-restricted-imports` rule in the provided example) in multiple overrides that match a file, only the configuration for that rule from the last override will apply. That is, ESLint does merge lists of rules (if you have different rules in two overrides, for example `no-restricted-imports` and `no-void`, all rules will apply), but it doesn't merge configurations for each rule.

**Reply by AnderssonPeter** [reply]
Is there any way to allow multiple instances of the same rule?

**Reply by mdjermanovic** [reply]
> Is there any way to allow multiple instances of the same rule?

ESLint runs each rule only once per file, so it isn't possible to run the same rule with two different configurations on the same file. A workaround could be to clone the rule (for example, copy & paste the code into a plugin rule) and then use the original rule and its copy as two different rules (`no-restricted-imports` and `myplugin/no-restricted-imports`).

**Reply by AnderssonPeter** [reply]
If you wish to earn 100 stackoverflow points post this plus a example at https://stackoverflow.com/questions/72184316/rules-are-only-applied-from-the-last-override
